lib/types.js

```diff
@@ -1663,10 +1670,13 @@ EnumType.prototype._skip = function (tap) { tap.skipLong(); };
 
 EnumType.prototype._write = function (tap, val) {
   var index = this._indices[val];
-  if (index === undefined) {
```

Comment:
If I understand the spec correctly, default enum symbols are only used during [schema resolution](http://avro.apache.org/docs/current/spec.html#Schema+Resolution) when a writer's symbol does not exist in the reader. We'll therefore need to modify `_update` rather than `_write`.

Instead of the `every` check that makes sure each writer symbol is in the reader, we could iterate over the writer's symbols and insert the reader's default if available when the symbol isn't found in the reader (rather than failing). Something like:

```js
EnumType.prototype._update = function (resolver, type) {
  if (
    type.typeName !== 'enum' ||
    (!type.name || ~getAliases(this).indexOf(type.name))
  ) {
    return;
  }
  var symbols = type.symbols.slice();
  var i, l, symbol;
  for (i = 0, l = symbols.length; i < l; i++) {
    symbol = symbols[i];
    if (this._indices[symbol] !== undefined) {
      continue; // Both the reader and the writer have this symbol.
    }
    if (this.defaultSymbol === undefined) {
      return; // Incompatible enums.
    }
    symbols[i] = this.defaultSymbol; // Substitute the default.
  }
  resolver.symbols = symbols;
  resolver._read = type._read; 
};
```

Reply:
Thank you for the feedback, I see what you mean now. It should be reflected in the PR now